tools/rosmsg/src/rosmsg/__init__.py

```diff
@@ -551,8 +551,13 @@ def _get_package_paths(pkgname, rospack):
     path = rospack.get_path(pkgname)
     paths.append(path)
     results = find_in_workspaces(search_dirs=['share'], project=pkgname, first_match_only=True, workspace_to_source_spaces=_catkin_workspace_to_source_spaces, source_path_to_packages=_catkin_source_path_to_packages)
-    if results and results[0] != path:
-        paths.append(results[0])
+    if results:
+        if os.path.sep != '/':
+            if path.replace(os.path.sep, '/') != results[0].replace(os.path.sep, '/'):
+                paths.append(results[0])
+        else:
+            if path_in_workspaces != path:
+                paths.append(results[0])
```

Comment:
The duplication of control seems unnecessary. This could simple stay a single condition since on platforms where `os.path.sep` is `/` the `replace()` calls are no-ops:

```
if results and results[0].replace(os.path.sep, '/') != path.replace(os.path.sep, '/'):
    paths.append(results[0])
````